llvm/include/llvm/IR/Module.h

```diff
@@ -234,6 +238,13 @@ class LLVM_EXTERNAL_VISIBILITY Module {
     IsNewDbgInfoFormat = false;
   }
 
+  void setIsNewDbgInfoFormat(bool NewFlag) {
+    if (NewFlag && !IsNewDbgInfoFormat)
+      convertToNewDbgValues();
+    else if (!NewFlag && IsNewDbgInfoFormat)
+      convertFromNewDbgValues();
```

Comment:
That looks a bit less readable to me, would it be better to change to this?
```
 if (IsNewFormat != IsNewDbgInfoFormat) {
        if (IsNewFormat) {
            convertToNewDebugFormat();
        } else {
            convertFromNewDebugFormat();
        }
     IsNewDbgInfoFormat = IsNewFormat;
   }
```
 `IsNewDbgInfoFormat = IsNewFormat;` // would be nice 
 We can then refactor/remove the `IsNewDbgInfoFormat = true/false` from `convertToNewDebugFormat()` and  `convertFromNewDbgValues()` and just keep this line in `setIsNewDbgInfoFormat` and then the method name also makes a bit more sense in terms of really "setting" the new debug format 😉

Reply:
I think it's important to keep `IsNewDbgInfoFormat = true` in the convert to/from methods - we aren't changing every conversion point to use this new method, and the flag needs to accurately reflect which format we're in at all times.

Reply:
Ah right, thanks for the explanation, still I think we can make `setIsNewDbgInfoFormat` a bit readable by
```
void setIsNewDbgInfoFormat(bool isNewFormat) {
    if (IsNewFormat == IsNewDbgInfoFormat) {
        return;
    }

    if (IsNewFormat) {
        convertToNewDebugFormat();
    } else {
        convertFromNewDebugFormat();
    }
}
```

Thought an early return make sense in terms of  if the requested format is the same as the current one

Reply:
Right now, I think I prefer the current approach primarily because it is consistent with the formatting of the equivalent function in `BasicBlock`, so is simpler to understand for someone who views both (rather than appearing to contain different logic) - obviously all these approaches are identical so I'm not attached to any one in particular, right now I just lean more towards the more concise and consistent approach rather than emphasizing simpler conditionals - WDYT?

Reply:
Ah Alright,it's fine! Thank You!